jekyll-kuma-plugins/lib/jekyll/kuma-plugins/liquid/tags/policyyaml.rb

```diff
@@ -2,26 +2,55 @@
 # It removes duplication of examples for both universal and kubernetes environments.
 # The expected format is universal. It only works for policies V2 with a `spec` blocks.
 require 'yaml'
+require 'rubygems'  # Required for Gem::Version
+
 module Jekyll
   module KumaPlugins
     module Liquid
       module Tags
         class PolicyYaml < ::Liquid::Block
+          TARGET_VERSION = Gem::Version.new("2.9.0")
+
           def initialize(tag_name, markup, options)
             super
             @tabs_name, *params_list = @markup.split(' ')
-            @params = {"raw" => false, "apiVersion" => "kuma.io/v1alpha1"}
+            @params = {"raw" => false, "apiVersion" => "kuma.io/v1alpha1", "use_meshservice" => "false" }
             params_list.each do |item|
                 sp = item.split('=')
                 @params[sp[0]] = sp[1] unless sp[1] == ''
             end
           end
 
-		  # process_hash and process_array are recursive functions that remove the suffixes from the keys
-		  # and rename the keys that have the suffixes.
-		  # For example, if you have keys called `name_uni` and `name_kube`:
-		  # on universal - `name_uni` -> `name` and `name_kube` will be removed
-		  # on kubernetes - `name_kube` -> `name` and `name_uni` will be removed
+          # Function to transform targetRef based on MeshService name (if needed)
+          def transform_target_ref(hash)
+            # Transform each element in spec.to[].targetRef.kind if spec.to exists
+            if hash.dig("spec", "to").is_a?(Array)
+              hash["spec"]["to"].each do |to_item|
+                if to_item.dig("targetRef", "kind") == "MeshService"
+                  target_ref = to_item["targetRef"]
+                  if target_ref["name_kube"]
+                    transformed_name = target_ref["name_kube"].split('_')
+                    to_item["targetRef"] = {
+                      "kind" => "MeshService",
+                      "name" => transformed_name[0],
+                      "namespace" => transformed_name[1],
```

Comment:
we shouldn't have namespace for universal policies, and this code will add it. We should check if this is for universal or k8s

Reply:
we talked offline and it will be addressed in #2003